### PR TITLE
ci: enable Vale (syntax checker) linter

### PR DESCRIPTION
**Pull request by jeromecambon**
The Vale linter checks the doc content and detect typos, incorrect syntax, and provide suggestions. See https://vale.sh for details.
Check [the video](https://github.com/bonitasoft/bonita-documentation-site/issues/162#issuecomment-1279060225) included in the issue for a demo!
Covers https://github.com/bonitasoft/bonita-documentation-site/issues/162
See also #2196

### CODE REVIEW

File: .github/workflows/contribution-check.yml [hunk 1]
```diff
@@ -17,9 +15,23 @@ jobs:
...
-        github-token: ${{ secrets.GITHUB_TOKEN }}
-        diffDoesNotContain: '["https://documentation.bonitasoft.com/", "Bonita BPM"]'
-        extensionsToCheck: '[".adoc"]'
+      - name: Check PR content
+        uses: bonitasoft/actions/packages/pr-diff-checker@1.1.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          diffDoesNotContain: '["https://documentation.bonitasoft.com/", "Bonita BPM"]'
+          extensionsToCheck: '[".adoc"]'
+      - uses: actions/checkout@v2
```
Comment (tbouffard) [inline .github/workflows/contribution-check.yml:24]: let's use v3

File: .github/workflows/contribution-check.yml [hunk 2]
```diff
@@ -17,9 +15,23 @@ jobs:
...
+
+      - id: files
```
Comment (tbouffard) [inline .github/workflows/contribution-check.yml:25]:
```suggestion
      - id: files
        name: Detect changed files
```

File: .github/workflows/contribution-check.yml [hunk 3]
```diff
@@ -17,9 +15,23 @@ jobs:
...
+        uses: jitterbit/get-changed-files@v1
+        with:
+          format: 'json'
+
```
Comment (tbouffard) [inline .github/workflows/contribution-check.yml]:
```suggestion
```

File: .github/styles/Vocab/Base/accept.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Bonitasoft
+Bonita
+Asciidoc
```
Comment (tbouffard) [inline .github/styles/Vocab/Base/accept.txt]: ⚠️ wrong syntax, see https://docs.asciidoctor.org/asciidoc/latest/
```suggestion
AsciiDoc
```

**Review by tbouffard — CHANGES_REQUESTED** [review]

File: .github/workflows/contribution-check.yml (hunk 1 above)
Comment (tbouffard) [inline .github/workflows/contribution-check.yml]:
```suggestion
      - uses: actions/checkout@v3
```

Comment (tbouffard) [reply .github/workflows/contribution-check.yml]: already applied

**Comment by tbouffard** [comment]
The dictionary is now OK. I don't have time to review the rules today. I wonder if there are any configured rules that could generate false positives.
Let's wait for someone else to review.

**Comment by jeromecambon** [comment]
> The dictionary is now OK. I don't have time to review the rules today. I wonder if there are any configured rules that could generate false positives.
> Let's wait for someone else to review.

The "dictionary" (accept.txt file) will evolve with time, it's just a start. We'll add new keywords when needed.
And yes, we'll have false positive. Typically when the page includes variable names, we can have many. So don't expect the check will be always green...

**Review by benjaminParisel — APPROVED** [review]
LGTM :clap:

**Review by tbouffard — APPROVED** [review]
> And yes, we'll have false positive. Typically when the page includes variable names, we can have many. So don't expect the check will be always green...

OK, that's clear. False positives are a pain and if we have too many, nobody will look at the results of the checks. Let's start with what is proposed here and see how it runs.

**Comment by tbouffard** [comment]
Put in Draft as we are waiting for feedback from the Product Team.

**Pull request merged**